### PR TITLE
[#652]UI: Button generator advanced options

### PR DESCRIPTION
**Pull request by johnkuney**
Related to #652

Description
---
Adding the advanced options to the button generator, as well as a toggle for the widget

Test plan
---
Go to the button generator and try the new options. Check the code output looks correct and the preview works

### CODE REVIEW

**Review by Klakurka — CHANGES_REQUESTED** [review]
Can't seem to build this branch.

image

**Comment by Klakurka** [comment]
Same thing after a git prune.

**Comment by johnkuney** [comment]
hmm maybe was behind some commits, just did a rebase

**Review by Klakurka — CHANGES_REQUESTED** [review]
Still getting the same error.

File: components/ButtonGenerator/CodeBlock.tsx [hunk 1]
```diff
@@ -115,12 +125,67 @@ export default function CodeBlock({ button }): JSX.Element {
     }
   };
 
+  const checkReactValue = (value, attribute) => {
+    if (value !== undefined && value !== '') {
+      if (attribute === 'currency') {
+        if (['USD', 'CAD', 'EUR', 'GBP', 'AUD'].includes(value)) {
```
Comment (Klakurka) [inline components/ButtonGenerator/CodeBlock.tsx:131]: I'd rather this come from a central list somewhere. Maybe from the DB. It'll be one of those things we miss when we add/remove currencies. In fact EUR, GPB, and AUD aren't even supported rn.

**Comment by johnkuney** [comment]
Will revisit this with a more incremental approach since we are having issues with the branch

**Pull request closed without merging**